pkg/mount/mount.go

```diff
@@ -479,8 +502,23 @@ func (m *Mounter) Mount(
 
 	// Record previous state of the path
 	pathWasReadOnly := m.isPathSetImmutable(path)
+	var (
+		isBindMounted bool = false
+		bindMountPath string
+	)
+
 	if err := m.makeMountpathReadOnly(path); err != nil {
-		return fmt.Errorf("failed to make %s readonly Err: %v", path, err)
+		if strings.Contains(err.Error(), "Inappropriate ioctl for device") {
```

Comment:
As discussed offline, not going to check whether it is mounted or not, as there could be an existing mount from the container orchestrator. This could happen only if an existing fuse mount is not removed. In that case we would get an ioctl error and we will mount on top the existing mount. So it will not be a problem even if there exists a mount.